portfolio/src/main/webapp/tasksApi.js

```diff
@@ -0,0 +1,193 @@
+  // Call the initClient function after the modules load.
+  gapi.load('client:auth2', initClientTasks);
+}
+
+function initClientTasks() {
+  fetchApiKey();
+
+  // Initialize the gapi.client object, which app uses to make API requests.
+  // Get API key and client ID from API Console.
+  // 'scope' field specifies space-delimited list of access scopes.
+  gapi.client
+      .init({
+        'apiKey': API_KEY_TASKS,
+        'clientId': CLIENT_ID_TASKS,
+        'discoveryDocs': [DISCOVERY_DOCS_TASKS],
+        'scope': SCOPE_READ_TASKS
+      })
+      .then(function() {
+        GoogleAuth = gapi.auth2.getAuthInstance();
+
+        // Listen for sign-in state changes.
+        GoogleAuth.isSignedIn.listen(updateSigninStatus);
+
+        // Handle initial sign-in state. (Determine if user is already signed
+        // in.)
+        updateSigninStatus(GoogleAuth.isSignedIn.get());
+      });
+}
+
+function handleAuthClick() {
+  if (GoogleAuth.isSignedIn.get()) {
+    // User is authorized and has clicked "Sign out" button.
+    GoogleAuth.signOut();
+  } else {
+    // User is not signed in. Start Google auth flow.
+    GoogleAuth.signIn();
+  }
+}
+
+function updateSigninStatus(isSignedIn) {
+  if (isSignedIn) {
+    drawImportMenu();
+  } else {
+    clearImportMenu();
+  }
+  updateButtonText(isSignedIn);
+}
+
+// Iterate through all the user's tasklists and pass
+// them to updateTaskList().
+function importAllTasks() {
+  if (GoogleAuth.isSignedIn.get()) {
+    gapi.client.tasks.tasklists.list({'maxResults': 100})
+        .then(function(response) {
+          var taskLists = response.result.items;
+          if (taskLists && taskLists.length > 0) {
+            for (var i = 0; i < taskLists.length; i++) {
+              var taskList = taskLists[i];
+              gapi.client.tasks.tasks
+                  .list({
+                    'tasklist': taskList.id,
+                    'maxResults': 100,
+                    'showCompleted': false
+                  })
+                  .then(function(taskResponse) {
+                    var tasks = taskResponse.result.items;
+                    if (tasks && tasks.length > 0) {
+                      for (var j = 0; j < tasks.length; j++) {
+                        task = tasks[j];
+
+                        const newTask =
+                            new Task(task.title, task.notes, '60', '3');
+                        updateTaskList(newTask, TIME_UNIT.MINUTES);
+                      }
+                    }
+                  });
+            }
+          }
+        });
+  }
+}
+
+// Import a single tasklist identified by its id.
+function importTasklist(tasklistId) {
+  gapi.client.tasks.tasks
+      .list({'tasklist': tasklistId, 'maxResults': 100, 'showCompleted': false})
+      .then(function(taskResponse) {
+        var tasks = taskResponse.result.items;
+        if (tasks && tasks.length > 0) {
+          for (var i = 0; i < tasks.length; i++) {
```

Comment:
This can probably be simplified into a for each loop.

Reply:
Done.